Document unavailable file behavior in release notes (#9952)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

For example, renaming an input of a `Copy` task called `file.txt` to `FILE.txt` will now cause `FILE.txt` being created in the destination directory. 
The `Sync` task and `Project.copy()` and `sync()` operations now also handle case-renames as expected.

## Unavailable files are handled more gracefully

Generally, broken symlinks, named pipes and unreadable files/directories (hereinafter referred to as unavailable files) found in inputs and outputs of tasks are handled gracefully from now on: as if they don't exist.

> For example copying into a directory with a leftover named pipe or broken symbolic link won't hang the build anymore.

Reviewer: ```suggestion
For example copying into a directory with a leftover named pipe or broken symbolic link won't break the build anymore.
```

The build hanged with a named pipe and failed with a symbolic link. I would go for `break`.

For example, renaming an input of a `Copy` task called `file.txt` to `FILE.txt` will now cause `FILE.txt` being created in the destination directory. 
The `Sync` task and `Project.copy()` and `sync()` operations now also handle case-renames as expected.

> ## Unavailable files are handled more gracefully

Reviewer: Can you add a newline after the header?

Author: Done

> Generally, broken symlinks, named pipes and unreadable files/directories (hereinafter referred to as unavailable files) found in inputs/outputs of tasks are handled gracefully from now on: similarly to missing files.

Reviewer: ```suggestion
Generally, broken symlinks, named pipes and unreadable files/directories (hereinafter referred to as unavailable files) found in inputs and outputs of tasks are handled gracefully from now on: as if they don't exist.
```

I don't think missing files is a term known to our users. If we want to make it so that they understand then we need to define this term in the user manual and link to it from here.

> 

Reviewer: Can we add an example use case here why this is useful? Or what is the change in behaviour?

Author: Added, if you think more is needed, please insert a suggestion.

> If the output contains an unavailable file - not produced by the task -, it will be ignored.

Reviewer: I think everything from here to the end can be removed.

Author: Okay, if you think so

For example, renaming an input of a `Copy` task called `file.txt` to `FILE.txt` will now cause `FILE.txt` being created in the destination directory. 
The `Sync` task and `Project.copy()` and `sync()` operations now also handle case-renames as expected.

## Unavailable files are handled more gracefully
Generally, broken symlinks, named pipes and unreadable files/directories (hereinafter referred to as unavailable files)

> found in inputs/outputs of tasks are handled gracefully from now on: similarly to missing files.

Reviewer: We should try to use one line per sentence, so it is easier to comment and track changes.

Author: That's what I tried, breaking longer sentences to multiple lines, it's what I saw in other places.
Do you suggest not breaking long sentences to multiple lines?

Author: Done

Reviewer: That is what I am suggesting. The user manual (or release notes) are not consistent, yet. We try to use the one sentence per line style when we touch or create something new, though.

> In incremental builds therefore, changing the availability of a file (e.g. modifying the target of a broken symlink to a valid file, or vice versa),

Reviewer: I think all this documentation should go to `more_about_tasks.adoc` in a new section talking about symbolic links under the `sec:up_to_date_checks`. The documentation about caching could go in the build cache guide under [symbolic links](https://guides.gradle.org/using-build-cache/#symbolic_links) or also in the up-to-date checks section.

WDYT?

Author: > The documentation about caching could go in the build cache guide

I understood that we should rather remove documenting the cache and the skipwhenempty parts, since it's confusing and will change in 6 anyway.

Author: Done, please check if it's what you've requested
